module/optee/mbx/test/mod_optee_mbx_unit_test.c

```diff
@@ -13,11 +13,14 @@
 
 #include <Mockfwk_id.h>
 #include <Mockfwk_module.h>
+#include <Mockfwk_mm.h>
+#include <Mockmod_optee_mbx_extra.h>
 
 #include <fwk_element.h>
 #include <fwk_macros.h>
 
 #include UNIT_TEST_SRC
+#include "config_optee_mbx.h"
 
 void setUp(void)
```

Comment:
Could you use setUp and tearDown functions to initialize the `mbx_ctx` and the other global variables?
This is to isolate the test side effects. For example, `test_optee_mbx_init` initializes the context and all these variables.
If something is wrong added to mbx_init, the test fails without correctly initializing the `ctx`s. Then, all the remaining tests will suffer from this.
To see the effect, you can change the order of the tests in `optee_mbx_test_main`.

Reply:
mbx_ctx is a optee/mbox module global variable that is initialized by mbx_init()

I overwrite some config to test the 2 type of shmem user.  I can probably create to context: one for static mem and one for dynamic instead

Reply:
I am not sure I understand this statement 
``` I overwrite some config to test the 2 type of shmem user. I can probably create to context: one for static mem and one for dynamic instead```
But my request is to get rid of the test side-effects on the global shared variables
The strategy used by the team so far is to have default values of these global variables. Then in every setup call, copy it to the actual variables used.

Example:
```
void setUp(void)
{
   //copy values of default_mbx_ctx to mbx_ctx
}
```

Speaking of which, It's advisable to use 
`Mock*_Destroy()` in the tearDown function If tests use mocks. This will help clear any pending 
mock expectation (It can happen if the test fails before the function(s) under test didn't call the mock function and fails for something else.)
You can ignore this for now, but it's worth mentioning.